streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -698,6 +707,9 @@ private long processAndPunctuateStreamTime(final Map<TaskId, StreamTask> tasks,
         } while (totalProcessedEachRound != 0);
 
         // go over the tasks again to punctuate or commit
+        // TODO: the thrown exception could be ProducerFenced, and hence need to be handled on the task that throws it
```

Comment:
Not a big fan of leaving `TODO`s around. Should we raise a JIRA instead? This seems like it is not something we want to forget. All to often `TODO`s never get done.